doc/scss/custom.scss

```diff
@@ -190,3 +190,54 @@ div.sk-text-image-grid-small {
 div.sk-text-image-grid-large {
   @include sk-text-image-grid(100px);
 }
+
+/* Style of the docsearch modal window */
+
+// In the following, we mainly reuse the background color and font-color from the
+// pydata-sphinx-theme. We also adjust slightly the font-size to be wider and in line
+// with the sidebar design.
+:root {
+  --docsearch-modal-width: 768px;
+  --docsearch-modal-height: 700px;
+  --docsearch-searchbox-shadow: inset 0 0 0 2px var(--pst-color-primary);
+  --docsearch-highlight-color: var(--pst-color-primary);
```

Comment:
So here I think that we have 2 potentials candidate as color:

- `--pst-color-primary`
- `--pst-color-accent`

The first one will be on the "blue-ish" side:

<img width="783" alt="image" src="https://github.com/scikit-learn/scikit-learn/assets/7454015/7490aa51-fcbb-4342-9634-28553881e01d">

The second one will be on the "pink-ish" side:

<img width="821" alt="image" src="https://github.com/scikit-learn/scikit-learn/assets/7454015/26f82b1e-a590-4314-8929-5be338f8fa1b">

This style is a bit more linked to the color used in the original theme when it comes to the search bar.

Reply:
I would vote for `--pst-color-primary` which IIRC is the color of links.

Reply:
Agreed. I now that I see the final rendering, I think that the blending is indeed better with the blue-ish style.